pkg/operators/kubeipresolver/kubeipresolver.go

```diff
@@ -134,6 +146,183 @@ func (m *KubeIPResolverInstance) EnrichEvent(ev any) error {
 	return nil
 }
 
+func (k *KubeIPResolver) GlobalParams() api.Params {
+	return nil
+}
+
+func (k *KubeIPResolver) InstanceParams() api.Params {
+	return nil
+}
+
+type endpointAccessors struct {
+	root            datasource.FieldAccessor
+	subK8sKind      datasource.FieldAccessor
+	subK8sName      datasource.FieldAccessor
+	subK8sNamespace datasource.FieldAccessor
+	subK8sLabels    datasource.FieldAccessor
+}
+
+func (k *KubeIPResolver) InstantiateDataOperator(gadgetCtx operators.GadgetContext, instanceParamValues api.ParamValues) (operators.DataOperatorInstance, error) {
+	k8sInventory, err := common.GetK8sInventoryCache()
+	if err != nil {
+		return nil, fmt.Errorf("creating k8s inventory cache: %w", err)
+	}
+	inst := &KubeIPResolverInstance{
+		k8sInventory: k8sInventory,
+	}
+
+	logger := gadgetCtx.Logger()
+	epAccessors := make(map[datasource.DataSource][]endpointAccessors)
+	for _, ds := range gadgetCtx.GetDataSources() {
+		logger.Debugf("KubeIPResolverOperator inspecting datasource %q", ds.Name())
+
+		endpoints := ds.GetFieldsWithTag("type:" + endpointL4Type)
+		if len(endpoints) > 0 {
+			logger.Debugf("> found %d endpoint fields", len(endpoints))
+			for _, ep := range endpoints {
+				k8sSubAcc, err := ep.AddSubField("k8s", api.Kind_Invalid, datasource.WithFlags(datasource.FieldFlagEmpty))
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "k8s", err)
+				}
+				k8sKindAcc, err := k8sSubAcc.AddSubField("kind", api.Kind_String, datasource.WithFlags(datasource.FieldFlagHidden))
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "kind", err)
+				}
+				k8sNameAcc, err := k8sSubAcc.AddSubField("name",
+					api.Kind_String,
+					datasource.WithAnnotations(map[string]string{
+						datasource.TemplateAnnotation: "pod",
+					}),
+					datasource.WithFlags(datasource.FieldFlagHidden),
+				)
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "name", err)
+				}
+				k8sNamespaceAcc, err := k8sSubAcc.AddSubField("namespace",
+					api.Kind_String,
+					datasource.WithAnnotations(map[string]string{
+						datasource.TemplateAnnotation: "namespace",
+					}),
+					datasource.WithFlags(datasource.FieldFlagHidden),
+				)
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "namespace", err)
+				}
+				k8sLabelsAcc, err := k8sSubAcc.AddSubField("labels", api.Kind_String, datasource.WithFlags(datasource.FieldFlagHidden))
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "labels", err)
+				}
+				ea := endpointAccessors{
+					root:            ep,
+					subK8sKind:      k8sKindAcc,
+					subK8sName:      k8sNameAcc,
+					subK8sNamespace: k8sNamespaceAcc,
+					subK8sLabels:    k8sLabelsAcc,
+				}
+				epAccessors[ds] = append(epAccessors[ds], ea)
+			}
+		}
+	}
+
+	inst.endpointsAccessors = epAccessors
+	return inst, nil
```

Comment:
Verify if any data source has `endpointL4Type` fields to avoid instantiating the operator for gadgets that don't need it:

```suggestion
	if len(epAccessors) == 0 {
		// Nothing to do
		return nil, nil
	}

	return &KubeIPResolverInstance{
		k8sInventory:       k8sInventory,
		endpointsAccessors: epAccessors,
	}, nil
```